share/scripts/inverse/prepare_generic_single.sh

```diff
@@ -34,32 +66,58 @@ main_dir=$(get_main_dir)
+    #resample target dist
+    do_external resample target "$(csg_get_interaction_property inverse.target)" "${name}.dist.tgt" 
+    # initial guess from rdf
+    raw="$(critical mktemp ${name}.pot.new.raw.XXX)"
+    kbt="$(csg_get_property cg.inverse.kBT)"
+    dist_min="$(csg_get_property cg.inverse.dist_min)"
+    do_external dist invert --type "${bondtype}" --kbT "${kbt}" --min "${dist_min}" ${name}.dist.tgt ${raw}
+    smooth="$(critical mktemp ${name}.pot.new.smooth.XXX)"
+    critical csg_resample --in ${raw} --out ${smooth} --grid ${min}:${step}:${max} --comment "${comment}"
+    extrapolate="$(critical mktemp ${name}.pot.new.extrapolate.XXX)"
+    do_external potential extrapolate --type "$bondtype" "${smooth}" "${extrapolate}"
+    shifted="$(critical mktemp ${name}.pot.new.shifted.XXX)"
+    do_external potential shift --type "${bondtype}" ${extrapolate} ${shifted}
+    do_external table change_flag "${shifted}" "${output}"
+}
+
+TABLE_PRESENT=false
 if [[ -f ${main_dir}/${name}.pot.in ]]; then
-  msg "Using given table ${name}.pot.in for ${name}"
-  smooth="$(critical mktemp ${name}.pot.in.smooth.XXX)"
-  echo "Converting ${main_dir}/${name}.pot.in to ${output}"
-  critical csg_resample --in "${main_dir}/${name}.pot.in" --out ${smooth} --grid ${min}:${step}:${max} --comment "$comment"
-  extrapolate="$(critical mktemp ${name}.pot.in.extrapolate.XXX)"
-  do_external potential extrapolate --type "$bondtype" "${smooth}" "${extrapolate}"
-  shifted="$(critical mktemp ${name}.pot.in.shifted.XXX)"
-  do_external potential shift --type "${bondtype}" ${extrapolate} ${shifted}
-  do_external table change_flag "${shifted}" "${output}"
-else
-  target=$(csg_get_interaction_property inverse.target)
-  msg "Using initial guess from dist ${target} for ${name}"
-  #resample target dist
-  do_external resample target "$(csg_get_interaction_property inverse.target)" "${name}.dist.tgt" 
-  # initial guess from rdf
-  raw="$(critical mktemp ${name}.pot.new.raw.XXX)"
-  kbt="$(csg_get_property cg.inverse.kBT)"
-  dist_min="$(csg_get_property cg.inverse.dist_min)"
-  do_external dist invert --type "${bondtype}" --kbT "${kbt}" --min "${dist_min}" ${name}.dist.tgt ${raw}
-  smooth="$(critical mktemp ${name}.pot.new.smooth.XXX)"
-  critical csg_resample --in ${raw} --out ${smooth} --grid ${min}:${step}:${max} --comment "${comment}"
-  extrapolate="$(critical mktemp ${name}.pot.new.extrapolate.XXX)"
-  do_external potential extrapolate --type "$bondtype" "${smooth}" "${extrapolate}"
-  shifted="$(critical mktemp ${name}.pot.new.shifted.XXX)"
-  do_external potential shift --type "${bondtype}" ${extrapolate} ${shifted}
-  do_external table change_flag "${shifted}" "${output}"
+    TABLE_PRESENT=true
 fi
 
+if [[ $use_bi == true ]]; then
+    if [[ $TABLE_PRESENT == true ]]; then
+        msg "there is a table ${name}.pot.in present, but you still choose BI"
+    fi
+    bi_init
+elif [[ $use_table == true ]]; then
+    if ! [[ $TABLE_PRESENT == true ]]; then
```

Comment:
```suggestion
    if [[ $TABLE_PRESENT != true ]]; then
```
plus lower case rename

Reply:
done